addons/hr_holidays/models/hr_leave.py

```diff
@@ -754,25 +756,35 @@ def _check_approval_update(self, state):
 
             if (state == 'validate1' and val_type == 'both') or (state == 'validate' and val_type == 'manager'):
                 manager = holiday.employee_id.parent_id or holiday.employee_id.department_id.manager_id
-                if (manager and manager != current_employee) and not self.env.user.has_group('hr_holidays.group_hr_holidays_manager'):
+                if not (holiday.employee_id.leave_manager_id and holiday.employee_id.leave_manager_id == self.env.user and is_team_leader) and (manager and manager != current_employee) and not is_manager:
                     raise UserError(_('You must be either %s\'s manager or Leave manager to approve this leave') % (holiday.employee_id.name))
 
             if state == 'validate' and val_type == 'both':
-                if not self.env.user.has_group('hr_holidays.group_hr_holidays_manager'):
+                if not is_manager:
                     raise UserError(_('Only an Leave Manager can apply the second approval on leave requests.'))
 
     # ------------------------------------------------------------
     # Activity methods
     # ------------------------------------------------------------
 
     def _get_responsible_for_approval(self):
-        if self.state == 'confirm' and self.employee_id.leave_manager_id:
-            return self.employee_id.leave_manager_id
-        elif self.state == 'confirm' and self.employee_id.parent_id.user_id:
-            return self.employee_id.parent_id.user_id
-        elif self.department_id.manager_id.user_id:
-            return self.department_id.manager_id.user_id
-        return self.env.user
+        self.ensure_one()
+        responsible = self.env.user
+
+        if self.validation_type == 'hr' or (self.validation_type == 'both' and self.state == 'validate1'):
+            responsible = self.env['res.users'].search([
```

Comment:
I find this really really weird to select someone randomly

Reply:
Yes I agree with you, this more than just weird this is just crappy

This feature is something we do in master using a field to specify who is the responsible directly on the leave type, but it seems that for some reason we only saw that this feature was apparently **absolutely essential** and that we could not live without it any longer so the basic plan is to implement the master's feature in saas-12.1.

Indeed trying to implement this feature in a stable version makes it that much harder since we cannot use a field (rightfully) hence the idea to just take someone at random and maybe at some point hardcode an id in internal. 
Which still seems like a bad idea to me, especially if the wait is only of about 5 months ...

Reply:
Yup, I don't see any other way to introduce this feature correctly in a stable version

Reply:
Completely offtopic to the PR yet about leaves: at the same time there's an accrual leave allocation that uses `create_date` of entry as when employee started working - if to talk about sanity ( ͡° ͜ʖ ͡°)